src/fontra_glyphs/backend.py

```diff
@@ -827,15 +838,49 @@ def variableGlyphToGSGlyph(defaultLocation, variableGlyph, gsGlyph, locationByMa
     fontraGlyphAxesToGSSmartComponentAxes(variableGlyph, gsGlyph)
     isSmartCompGlyph = len(gsGlyph.smartComponentAxes) > 0
 
-    nonSourceLayerNames = set(variableGlyph.layers)
-    for glyphSource in variableGlyph.sources:
-        layerName = glyphSource.layerName
-        nonSourceLayerNames.discard(layerName)
+    nonSourceLayerNames = set()
+    layerNameToGlyphSourceMapping = {
+        glyphSource.layerName: glyphSource for glyphSource in variableGlyph.sources
+    }
+    for layerName in variableGlyph.layers:
         layer = variableGlyph.layers[layerName]
-        gsLayer = gsGlyph.layers[layerName]
+        # "^" not in layerName:
         # layerName is equal to gsLayer.layerId if it comes from Glyphsapp,
         # otherwise the layer has been newly created within Fontra.
 
+        layerNameDescriptor = None
+        associatedMasterId = None
+        if "^" in layerName:
+            # Example: <parent-layer-name>^<background-layer-name>
+            # Example: <masterId>^background
+            glyphSourceLayerName, layerNameDescriptor = layerName.split("^")
```

Comment:
Ensure you split only once, just in case there are multiple `^` chars in the string:
```suggestion
            glyphSourceLayerName, layerNameDescriptor = layerName.split("^", 1)
```